### DISCUSSION TITLE
Primo tentativo integrazione

### DISCUSSION THREAD

**Discussion by ilguardiano** [fattureincloud/api#320] [Q&A] [answered]
Buongiorno, sto tentando oggi di iniziare ad implementare una integrazione con Fattureincloud e stavo seguendo il manuale per integrare le api.
Ho configurato una autenticazione manuale ricevendo un token.
Da qui, ho cercato di estrarre il companyid ed utilizzando l'sdk (con l'esempio copiaincollato)

**private void ProvaCompanyId ()
{
    Configuration config = new Configuration();
    config.BasePath = "https://api-v2.fattureincloud.it";
    // Check out the Authentication section to retrieve the Access Token
    config.AccessToken = MyToken;

    var apiInstance = new UserApi(config);

    try
    {
        // List User Companies
        ListUserCompaniesResponse result = apiInstance.ListUserCompanies();
        // In the result you'll find a list of companies, you must use the "Id" field of one of those
        Console.Write(result.Data.Companies[0].Id);
    }
    catch (ApiException e)
    {
        Console.Write("Exception when calling UserApi.ListUserCompanies: " + e.Message);
        Console.Write("Status Code: " + e.ErrorCode);
        Console.Write(e.StackTrace);
    }
}**
 e mi da questo errore:

Method not found: 'RestSharp.RestRequest RestSharp.RestRequestExtensions.AddFile(RestSharp.RestRequest, System.String, Byte[], System.String, System.String)'.

se lancio la singola rest api, da codice o da postman funziona invece

ma che sto sbagliando?

**Comment by emajo** [comment]
Ciao @ilguardiano l'errore sembrerebbe essere dovuto a un conflitto di dipendenze, potrebbe essere utile sapere la versione di restsharp installata e la versione .net.
Questo [thread di stackoverflow](https://stackoverflow.com/questions/47678696/method-not-found-restsharp-irestreuqest-restsharp-restrequest-addfile) espone un problema simile, forse può essere d'aiuto.

Grazie,
Emanuele.

**Reply by ilguardiano** [reply]
Ho guardato il mio progetto ha l'ultima versione di restsharp ovvero 110.2.0 mentre l'sdk ha la 108.0.3, ma non so se è questo.
Secondo me sbaglio qualcosa nella configurazione.
Io ho usato un autenticazione manuale ed ho messo il token in accesstoken. (del config). E' corretto? perchè mi pare di capire che deve essere usato per OAuth2 authentication....e se si, come faccio a mettere il mio token nella configurazione?

grazie

**Reply by emajo** [reply]
Si, è corretto impostare il token in quel modo, potrebbe trattarsi della versione di restsharp, sarebbe utile fare un tentativo con la 108 in quanto può darsi che la libreria sia cambiata nelle versioni successive.

Grazie,
Emanuele.

**Comment by ilguardiano** [comment]
Beh, per lo meno ora ha cambiato l'errore: 
### The format of value 'FattureInCloud/2.0.19/csharp-SDK' is invalid

**Reply by emajo** [reply]
Stiamo indagando per risolvere questo problema, per ora la soluzione è impostare manualmente lo user agent in questo modo
```
 Configuration config = new Configuration();
 config.UserAgent = "FattureInCloudSdk";
```

**Reply by ilguardiano** [reply]
Grazie, aggiungendo quel parametro ora funziona.
Non potreste anche aggiornare il riferimento all'ultima versione di RestSharp?

Grazie

**Reply by emajo** [reply]
Andremo a rilasciare a breve una nuova versione della libreria per risolvere questo problema dello user agent, per quanto riguarda la versione di rest sharp richiede di modificare una buona parte del codice ed andremo a farlo in futuro.

Grazie,
Emanuele.

**Reply by ilguardiano** [reply]
Grazie...allora aspetto !!!

**Reply by emajo** [reply, marked as answer]
Abbiamo rlasciato ieri la versione 2.0.21 che va a risolvere il problema dello user agent.

Grazie,
Emanuele.

**Reply by valmoz** [reply]
Buongiorno @ilguardiano 
il problema è stato risolto?

Grazie
Mauro

**Comment by ilguardiano** [comment]
il problema è stato risolto.. scusate il ritardo nella risposta, anche se ora ho un altro problema
